ios/AppLovinMAXAdView.m

```diff
@@ -0,0 +1,190 @@
+#import <AppLovinSDK/AppLovinSDK.h>
+#import "AppLovinMAX.h"
+#import "AppLovinMAXAdView.h"
+
+@interface AppLovinMAXAdView()
+
+@property (nonatomic, strong, nullable) MAAdView *adView; // nil when unmounted
+
+// The following properties are updated from RN layer via the view manager
+@property (nonatomic, copy) NSString *adUnitId;
+@property (nonatomic, weak) MAAdFormat *adFormat;
+@property (nonatomic, copy, nullable) NSString *placement;
+@property (nonatomic, copy, nullable) NSString *customData;
+@property (nonatomic, assign, readonly, getter=isAdaptiveBannerEnabled) BOOL adaptiveBannerEnabled;
+@property (nonatomic, assign, readonly, getter=isAutoRefresh) BOOL autoRefresh;
+
+@end
+
+@implementation AppLovinMAXAdView
+
+- (void)setAdUnitId:(NSString *)adUnitId
+{
+    // Ad Unit ID must be set prior to creating MAAdView
+    if ( self.adView )
+    {
+        [AppLovinMAX.shared log: @"Attempting to set Ad Unit ID %@ after MAAdView is created", adUnitId];
+        return;
+    }
+    
+    _adUnitId = adUnitId;
+    
+    [self attachAdView];
+}  
+
+- (void)setAdFormat:(NSString *)adFormat
+{
+    // Ad format must be set prior to creating MAAdView
+    if ( self.adView )
+    {
+        [AppLovinMAX.shared log: @"Attempting to set ad format %@ after MAAdView is created", adFormat];
+        return;
+    }
+    
+    if ( [@"banner" isEqualToString: adFormat] )
+    {
+        _adFormat = DEVICE_SPECIFIC_ADVIEW_AD_FORMAT;
+    }
+    else if ( [@"mrec" isEqualToString: adFormat] )
+    {
+        _adFormat = MAAdFormat.mrec;
+    }
+    else
+    {
+        [AppLovinMAX.shared log: @"Attempting to set an invalid ad format of \"%@\" for %@", adFormat, self.adUnitId];
+        return;
+    }
+    
+    [self attachAdView];
+}  
+
+- (void)setPlacement:(NSString *)placement
+{
+    _placement = placement;
+    
+    if ( self.adView )
+    {
+        self.adView.placement = placement;
+    }
+}
+
+- (void)setCustomData:(NSString *)customData
+{
+    _customData = customData;
+    
+    if ( self.adView )
+    {
+        self.adView.customData = customData;
+    }
+}
+
+- (void)setAdaptiveBannerEnabled:(BOOL)adaptiveBannerEnabled
+{
+    _adaptiveBannerEnabled = adaptiveBannerEnabled;
+    
+    if ( self.adView )
+    {
+        [self.adView setExtraParameterForKey: @"adaptive_banner" value: adaptiveBannerEnabled ? @"true" : @"false"];
+    }
+}
+
+- (void)setAutoRefresh:(BOOL)autoRefresh
+{
+    _autoRefresh = autoRefresh;
+    
+    if ( self.adView )
+    {
+        if ( autoRefresh )
+        {
+            [self.adView startAutoRefresh];
+        }
+        else
+        {
+            [self.adView stopAutoRefresh];
+        }
+    }
+}
+
+- (void)attachAdView
```

Comment:
nit but we should rename this to `attachAdViewIfNeeded` - this follows general Apple/iOS conventions for methods that can perform an action based on certain rules. (the conventions they put forth is to suffix with `isNeeded`).

On Android -> `maybeAttachAdView()` - this is more of our own convention as we've seen different styles/conventions in AOSP.

Reply:
Thanks, updated.